clippy_lints/src/methods/mod.rs

```diff
@@ -1020,11 +1020,11 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Methods {
             ["map", "find"] => lint_find_map(cx, expr, arg_lists[1], arg_lists[0]),
             ["flat_map", "filter"] => lint_filter_flat_map(cx, expr, arg_lists[1], arg_lists[0]),
             ["flat_map", "filter_map"] => lint_filter_map_flat_map(cx, expr, arg_lists[1], arg_lists[0]),
-            ["flat_map", ..] => lint_flat_map_identity(cx, expr, arg_lists[0]),
+            ["flat_map", ..] => lint_flat_map_identity(cx, expr, arg_lists[0], method_spans[0]),
             ["flatten", "map"] => lint_map_flatten(cx, expr, arg_lists[1]),
-            ["is_some", "find"] => lint_search_is_some(cx, expr, "find", arg_lists[1], arg_lists[0]),
-            ["is_some", "position"] => lint_search_is_some(cx, expr, "position", arg_lists[1], arg_lists[0]),
-            ["is_some", "rposition"] => lint_search_is_some(cx, expr, "rposition", arg_lists[1], arg_lists[0]),
+            ["is_some", "find"] => lint_search_is_some(cx, expr, "find", arg_lists[1], arg_lists[0], method_spans[0]),
```

Comment:
```suggestion
            ["is_some", "find"] => lint_search_is_some(cx, expr, "find", arg_lists[1], arg_lists[0], method_spans[1]),
```

the method/span lists are sorted with the most recent call first.

Reply:
Bonus points if you add this to the documentation of the `utils::method_calls` function :smile:

Reply:
the code suggestion is still wrong :man_shrugging:

Reply:
The span is correct now and thanks for adding the note to the docs :tada:

The replacement of the suggestion doesn't work, for some reason. Also I don't think that the `tests/ui/methods.rs` file is ready that `run-rustfix` is possible. You can either leave this comment out or extract the tests for `search_is_some` in another file.

It would be great if you could use the `get_arg_name` function I mentioned [above](https://github.com/rust-lang/rust-clippy/pull/4454#issuecomment-525639427)